### PR TITLE
Enable TLS Configuration for gRPC server with keystore file

### PR DESCRIPTION
**Pull request by EuroLew**
## Description

This will also configure the gRPC server with TLS using the keystore (PKCS12 format) file, there has also been minor refactoring since both the broker TLS config and gateway TLS config share code (in that they both validate the config to be mutually exclusive between the new keystore file and old key + cert chain file, and both require retrieval of private keys and cert chains from the keystore file when configuring the ssl context for netty).

## Related issues

relates to https://github.com/camunda/issues/issues/833

### CODE REVIEW

File: zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java
Comment (npepinpe) [inline zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java:32]: :wrench: Something I missed last time, we have some conventions for error messages: https://github.com/camunda/camunda/wiki/Error-Guidelines

So this would be reworded to something like `Expected to configure with a certificate and private key pair, or with a key store and password, but both were provided. Please select only one approach.`

File: zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java
Comment (npepinpe) [inline zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java:20]: :wrench: Just a heads up, since utility classes aren't meant to be instantiated, you would typically also make it final, and provide a private default constructor. This makes the intent of the class clearer (i.e. just a collection of utility methods).

**Review by npepinpe — APPROVED** [review]
:rocket: No need for another review, but please address comments before merging.

- :clown_face: I'm not big on utility classes in general, and I wish we would've found a better way to share this configuration, especially since a lot of the properties themselves are also duplicated/shared. But I also don't have a really strong alternative to suggest OTOH, so let's go with that. 
- :question: I forgot to ask, can I use a passwordless key store with this? I assume not right now, and it might create a NullPointerException, right? If effort is < 2 hours to add support for passwordless, then let's do it as a follow up PR. If not, then let's at least not throw a NullPointerException but instead a proper exception if no password is given in the validation. :thinking:

File: zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java
Comment (EuroLew) [reply zeebe/util/src/main/java/io/camunda/zeebe/util/TlsConfigUtil.java:21]: done

**Comment by EuroLew** [comment]
> 🚀 No need for another review, but please address comments before merging.
> 
> * 🤡 I'm not big on utility classes in general, and I wish we would've found a better way to share this configuration, especially since a lot of the properties themselves are also duplicated/shared. But I also don't have a really strong alternative to suggest OTOH, so let's go with that.
> * ❓ I forgot to ask, can I use a passwordless key store with this? I assume not right now, and it might create a NullPointerException, right? If effort is < 2 hours to add support for passwordless, then let's do it as a follow up PR. If not, then let's at least not throw a NullPointerException but instead a proper exception if no password is given in the validation. 🤔

Agree ideally the configs would get merged but for now its a stop gap (hopefully). Also now supports passwordless keystores

**Pull request merged**
